tests/test_echo.py

```diff
@@ -2,11 +2,160 @@
 # -*- coding: utf-8 -*-
 
 import unittest
+import subprocess
 import echo
 
+
 # Your test case class goes here
 class TestEcho(unittest.TestCase):
-    pass
+    def setUp(self):
+        self.parser = echo.create_parser()
+
+    def test_help(self):
+        """ Running the program without arguments should show usage. """
+
+        # Run the command `python ./echo.py -h` in a separate process, then
+        # collect its output.
+        process = subprocess.Popen(
+            ["python", "./echo.py", "-h"],
+            stdout=subprocess.PIPE)
+        stdout, _ = process.communicate()
+        stdout = stdout.decode("utf-8")
+        usage = open("./USAGE", "r").read()
+        self.assertEquals(stdout, usage)
+
+    def test_upper(self):
```

Comment:
I noticed that when my tests failed, the docstring of the failing test would print out -- but it got cut off after a certain number of characters. So I refactored these docstrings to make them shorter.

Reply:
Good thinking!